joplin/base/views/joplin_search_views.py

```diff
@@ -23,19 +25,49 @@
  - excluding certain pages from the pages to query
 """
 
+
+def dept_explorable_pages(user):
+    '''
+    In the spirit of UserPagePermissionsProxy's explorable_pages, check a user's permissions, excluding editor or
+    moderator permissions since those cast a wide net
+    https://github.com/wagtail/wagtail/blob/982b1d60a4b3d5b4e841f2b39e2244bcb421b091/wagtail/core/models.py#L1853
+    :param user: user making request
+    :return: Queryset of explorable pages
+    '''
+    # Deal with the trivial cases first...
+    if not user.is_active:
+        return Page.objects.none()
+    if user.is_superuser:
+        return Page.objects.all()  # todo: check this, we don't want the home page returned
+
+    user_perms = UserPagePermissionsProxy(user)
+    explorable_pages = Page.objects.none()
+
+    # exclude the moderator and editor groups
+    for perm in user_perms.permissions.filter(~Q(group_id=1) & ~Q(group_id=2)):
+        explorable_pages |= Page.objects.descendant_of(
+            perm.page, inclusive=True
+        )
```

Comment:
So this is interesting. It's looking at each permission that a user has. And then it's getting all the pages that share that permission? So if a user ever belong to multiple departments, we could iterate through the pages that match those permissions. Is that right?

Reply:
user_perms.permissions is a QuerySet of GroupPagePermissions ala: 
```
>>> up.permissions
<QuerySet [<GroupPagePermission: Group 2 ('Editors') has permission 'edit' on page 1 ('')>, <GroupPagePermission: Group 2 ('Editors') has permission 'add' on page 1 ('')>, <GroupPagePermission: Group 3 ('Kitchen sink department') has permission 'edit' on page 32 ('kitchen info')>, <GroupPagePermission: Group 3 ('Kitchen sink department') has permission 'edit' on page 14 ('Kitchen sink service page [en]')>]>
```
I expect that if a page has multiple departments with it, it will have GroupPagePermissions that cover both groups. Am I right @briaguya ?

Reply:
yes, a page under multiple departments is really just multiple GroupPagePermission entries for each department.